### PR TITLE
feat(shared): BlockItemWrapper allow undefined toolbar items

### PR DESCRIPTION
**Pull request by celine-s**

### CODE REVIEW

File: packages/shared/src/components/BlockInjectButton/types.ts
Comment (ragi96) [inline packages/shared/src/components/BlockInjectButton/types.ts:1]: These changes are not related to the BlockItemWrapper and are belonging to #539

Comment (celine-s) [reply packages/shared/src/components/BlockInjectButton/types.ts:1]: uuups, thanks! Removed

**Review by ragi96 — APPROVED** [review]

File: packages/shared/src/components/BlockItemWrapper/types.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { MenuItemStyle } from '@frontify/fondue';
 export type BlockItemWrapperProps = {
     shouldHideWrapper?: boolean;
     shouldHideComponent?: boolean;
-    toolbarItems: ToolbarItem[];
+    toolbarItems: (ToolbarItem | undefined)[];
```
Comment (fulopdaniel) [inline packages/shared/src/components/BlockItemWrapper/types.ts:8]: Shouldn't providing props in the right format would be the responsibility of the consumer? Where is this used where we pass undefined values inside the array?

Comment (celine-s) [reply packages/shared/src/components/BlockItemWrapper/types.ts:8]: Fair enough, I see your point. It is e.g. in the Thumbnail Grid needed in the following scenario: If there is less than 2 items I don't want to show the drag and Drop handle. Therefore, I supposed also in other blocks it might be usefull to just pass undefined instead of handling the filtering in each block. We can also do it in the block. Or wdyt?

**Review by fulopdaniel — APPROVED** [review]

File: packages/shared/src/components/BlockItemWrapper/types.ts (hunk 1 above)
Comment (SamuelAlev) [reply packages/shared/src/components/BlockItemWrapper/types.ts:8]: Daniel is actually right, the responsability is on the components to not pass `undefined`
